sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -201,6 +210,10 @@ static final DataType getSparkType(DType type) {
         return DataTypes.TimestampType;
       case STRING:
         return DataTypes.StringType;
+      case DECIMAL32:
+        return new DecimalType(DType.DECIMAL32_MAX_PRECISION, -type.getScale());
```

Comment:
Do we need to track the effective precision of the value as it progresses through operations in order to be compatible with how Spark operates on `Decimal`?  For example, take this simple Spark shell session:
```scala
scala> val x = Decimal(5.43)
x: org.apache.spark.sql.types.Decimal = 5.43

scala> (x.precision, x.scale)
res2: (Int, Int) = (3,2)

scala> val y = x * x
y: org.apache.spark.sql.types.Decimal = 29.4849

scala> (y.precision, y.scale)
res3: (Int, Int) = (6,4)
```

If we perform the same operation with `DECIMAL32` we will lose track of the precision, and when we cast back to a SparkType instead of getting `DecimalType(6,4)` as we should, we'll get `DecimalType(9,4)` instead.

Reply:
Spark should already be doing that for us, which is another reason why I don't think we want to rely on this API much longer.  We are in the process of moving away from it for nested types and I think it would be best to just not support it for decimal at all.

Reply:
Got it. I'll remove decimal parts.

Reply:
@revans2 @jlowe  The problem is that `GpuColumnVector.from(cudf.ColumnVector cudfCv)` relies on the method `getSparkType`. If `getSparkType` doesn't support DecimalTypes, we won't be able to create decimal-typed `GpuColumnVector` from `cudf.ColumnVector` through the method `GpuColumnVector.from(cudf.ColumnVector cudfCv)`.

Reply:
I am in the process of updating the plugin so `GpuColumnVector.from(cudf.ColumnVector cudfCv)` goes away and is replaced with `GpuColumnVector.from(cudf.ColumnVector cudfCv, DataType)`, similar to what I have been doing for `Table` to `ColumnarBatch` conversion. I have a patch working, just need to finish testing it and a bit more cleanup.  Because no one has taken a look at https://github.com/NVIDIA/spark-rapids/pull/1078 yet, I'll likely just update it instead of making a separate PR.

The goal eventually is to have `getSparkType` go away entirely because we cannot support it when there is not a 1:1 mapping between cudf types and spark types.

Reply:
I've refined this PR on the basis of the patch #1078. I appended decimal type check into `typeConversionAllowed`.